src/app/clients/edit-client/edit-client.component.html

```diff
@@ -109,20 +121,20 @@
 
           <mat-form-field fxFlex="48%">
             <mat-label>Submitted On</mat-label>
-            <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="submittedOnDatePicker" formControlName="submittedOnDate">
+            <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="submittedOnDatePicker" formControlName="submittedOnDate" required>
             <mat-datepicker-toggle matSuffix [for]="submittedOnDatePicker"></mat-datepicker-toggle>
             <mat-datepicker #submittedOnDatePicker></mat-datepicker>
+            <mat-error *ngIf="editClientForm.controls.submittedOnDate.hasError('required')">
+              Submitted Date is <strong>required</strong>
+            </mat-error>
           </mat-form-field>
 
           <mat-form-field fxFlex="48%">
             <mat-label>Activated On</mat-label>
-            <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="activatedOnDatePicker" required
+            <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="activatedOnDatePicker"
```

Comment:
Same here.

Reply:
In the edit case, If the client has not been activated yet why this field must be required. It's clear when the client is already activated It must be required.